### PR TITLE
fix: Don't include duplicates for SHOW DATABASES

### LINKED ISSUE #19984: SHOW DATABASES returns duplicates
## What

When there are multiple DB/RP mappings with the same database name, `SHOW DATABASES` incorrectly returns duplicates. Note `telegraf` appears twice in the following results:

```json
{
  "results": [
    {
      "statement_id": 0,
      "series": [
        {
          "name": "databases",
          "columns": [
            "name"
          ],
          "values": [
            [
              "chronograf"
            ],
            [
              "potato"
            ],
            [
              "potatoctl"
            ],
            [
              "telegraf"
            ],
            [
              "telegraf"
            ]
          ]
        }
      ]
    }
  ]
}
```

This results in duplicates in Chronograf 1.x:

### PR DESCRIPTION
**Pull request by stuartcarnie**
Closes #19984 

This PR ensures correct behaviour of `SHOW DATABASES`, such that it does not return duplicates for the same database name. Unit tests were added to verify this expectation.

Requires #20004 

- [x] [CHANGELOG.md](https://github.com/influxdata/influxdb/blob/master/CHANGELOG.md) updated with a link to the PR (not the Issue)
- [x] [Well-formatted commit messages](https://www.conventionalcommits.org/en/v1.0.0-beta.3/)
- [x] Rebased/mergeable
- [x] Tests pass

### CODE REVIEW

File: influxql/_v1tests/server_helpers.go [hunk 1]
```diff
@@ -146,37 +154,32 @@ func (qt *Test) addQueries(q ...*Query) {
...
-	writeOrg, err := influxdb.NewPermissionAtID(qt.orgID, influxdb.WriteAction, influxdb.OrgsResourceType, qt.orgID)
-	require.NoError(t, err)
-
-	bucketWritePerm, err := influxdb.NewPermissionAtID(qt.bucketID, influxdb.WriteAction, influxdb.BucketsResourceType, qt.orgID)
-	require.NoError(t, err)
-
-	bucketReadPerm, err := influxdb.NewPermissionAtID(qt.bucketID, influxdb.ReadAction, influxdb.BucketsResourceType, qt.orgID)
-	require.NoError(t, err)
+	if !qt.noWrites {
+		require.GreaterOrEqual(t, len(qt.writes), 0)
```
Comment (danxmoran) [inline influxql/_v1tests/server_helpers.go:163]: This was `require.Greater` before, is the change to `GreaterOrEqual` intentional?

**Review by danxmoran — APPROVED** [review]
Fix and test LGTM, one question about the helpers

File: influxql/_v1tests/server_helpers.go (hunk 1 above)
Comment (stuartcarnie) [reply influxql/_v1tests/server_helpers.go:163]: No, I initially wasn't going to have a `noWrites` flag, and was testing `GreaterOrEqual`, but realised that defeated the purpose, as writes will always be ≥ 0 🤦

**Review by danxmoran — APPROVED** [review]

**Review by danxmoran — APPROVED** [review]
:shipit: thanks for fixing the build!

**Pull request merged**